src/xo-app/host/tab-network.js

```diff
@@ -89,163 +89,223 @@ class ConfigureIpModal extends Component {
-            onChange={() => editNetwork(pif.$network, { defaultIsLocked: !networks[pif.$network].defaultIsLocked })}
-            value={networks[pif.$network].defaultIsLocked}
-          />,
-          pifInUse && _('pifInUse')
-        )}
-      </td>
-      <td>
-        <StateButton
-          disabledLabel={_('pifDisconnected')}
-          disabledHandler={connectPif}
-          disabledTooltip={_('connectPif')}
-
-          enabledLabel={_('pifConnected')}
-          enabledHandler={disconnectPif}
-          enabledTooltip={_('disconnectPif')}
-
-          disabled={pif.attached && (pif.management || pif.disallowUnplug)}
-          handlerParam={pif}
-          state={pif.attached}
+const COLUMNS = [
+  {
+    itemRenderer: pif => pif.device,
+    default: true,
+    name: _('pifDeviceLabel'),
+    sortCriteria: 'device'
+  },
+  {
+    itemRenderer: (pif, networks) => networks[pif.$network].name_label,
+    name: _('pifNetworkLabel'),
+    sortCriteria: (pif, networks) => networks[pif.$network].name_label
+  },
+  {
+    component: PifItemVlan,
+    name: _('pifVlanLabel'),
+    sortCriteria: 'vlan'
+  },
+  {
+    itemRenderer: (pif, networks) => <PifItemIp pif={pif} networks={networks} />,
+    name: _('pifAddressLabel'),
+    sortCriteria: 'ip'
+  },
+  {
+    itemRenderer: (pif, networks) => <PifItemMode pif={pif} networks={networks} />,
+    name: _('pifModeLabel'),
+    sortCriteria: 'mode'
+  },
+  {
+    itemRenderer: pif => pif.mac,
+    name: _('pifMacLabel'),
+    sortCriteria: 'mac'
+  },
+  {
+    itemRenderer: pif => pif.mtu,
+    name: _('pifMtuLabel'),
+    sortCriteria: 'mtu'
+  },
+  {
+    itemRenderer: (pif, networks) => <PifItemControl pif={pif} networks={networks} />,
+    name: _('defaultLockingMode')
+  },
+  {
+    itemRenderer: pif => <div>
+      <StateButton
+        disabledLabel={_('pifDisconnected')}
+        disabledHandler={connectPif}
+        disabledTooltip={_('connectPif')}
+
+        enabledLabel={_('pifConnected')}
+        enabledHandler={disconnectPif}
+        enabledTooltip={_('disconnectPif')}
+
+        disabled={pif.attached && (pif.management || pif.disallowUnplug)}
+        handlerParam={pif}
+        state={pif.attached}
+      />
+      {' '}
+      <Tooltip content={pif.carrier ? _('pifPhysicallyConnected') : _('pifPhysicallyDisconnected')}>
+        <Icon
+          icon='network'
+          size='lg'
+          className={pif.carrier ? 'text-success' : 'text-muted'}
         />
-        {' '}
-        <Tooltip content={pif.carrier ? _('pifPhysicallyConnected') : _('pifPhysicallyDisconnected')}>
-          <Icon
-            icon='network'
-            size='lg'
-            className={pif.carrier ? 'text-success' : 'text-muted'}
+      </Tooltip>
+    </div>,
+    name: _('pifStatusLabel')
+  }
+]
+
+const INDIVIDUAL_ACTIONS = [
+  {
+    handler: deletePif,
+    icon: 'delete',
+    label: _('deletePif')
+  }
+]
+
+const GROUPED_ACTIONS = [
+  {
+    label: 'deletePifs',
```

Comment:
This one also needs to be a node, right?

Reply:
Yes, you're right.